spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelFactory.java

```diff
@@ -25,5 +25,7 @@
  */
 public interface ChannelFactory {
```

Comment:
the methods here could be parameterized, e.g.:

```
<T extends MessageChannel> T createBindableChannel(String name, Class<T> channelType) throws Exception;

<T extends MessageChannel> T createSharedChannel(Class<T> channelType) throws Exception;
```

That would make it unnecessary to cast as in https://github.com/spring-cloud/spring-cloud-stream/pull/219/files#diff-d2113fc67655244ed5f8c586d6302862R130